addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/handler/NeatoHandler.java

```diff
@@ -0,0 +1,257 @@
+
+        super.initialize();
+
+        try {
+            refreshTime = ((BigDecimal) config.get(CONFIG_REFRESHTIME)).intValue();
+            if (refreshTime <= 0) {
+                throw new IllegalArgumentException("Refresh time must be positive number!");
+            }
+        } catch (Exception e) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}. {}", refreshTime,
+                    DEFAULTREFRESHTIME, e.getMessage());
+            refreshTime = DEFAULTREFRESHTIME;
+        }
+
+        mrRobot = new NeatoRobot(vacuumSerialNumber, vacuumSecret, vacuumName);
+        try {
+
+            this.startAutomaticRefresh();
+
+        } catch (Exception ex) {
+            // Note: When initialization can NOT be done set the status with more details for further
+            // analysis. See also class ThingStatusDetail for all available status details.
+            // Add a description to give user information to understand why thing does not work
+            // as expected. E.g.
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+            // "Can not access device as username and/or password are invalid");
+
+            logger.error(ex.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+            thisStatus = ThingStatus.OFFLINE;
+
+        }
+
+    }
+
+    public void refreshStateAndUpdate() {
+        try {
+
+            if (mrRobot.sendGetState()) {
+
+                if (thisStatus != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                    thisStatus = ThingStatus.ONLINE;
+                }
+
+                mrRobot.sendGetGeneralInfo();
+
+                List<Channel> channels = getThing().getChannels();
+
+                for (Channel channel : channels) {
+                    publishChannelIfLinked(channel.getUID());
+                }
+
+            }
+        } catch (Exception e) {
+            logger.error("Error when refreshing state. Error: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            thisStatus = ThingStatus.OFFLINE;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    refreshStateAndUpdate();
+
+                } catch (Exception e) {
+                    logger.error("Error starting automatic refresh. Erorr: {}", e.getMessage());
+                }
+
+            }
+        };
+
+        this.refreshTask = scheduler.scheduleAtFixedRate(refresher, 0, refreshTime, TimeUnit.SECONDS);
+        logger.debug("Start automatic refresh at {} minutes", refreshTime);
+    }
+
+    private void publishChannelIfLinked(ChannelUID channelUID) {
```

Comment:
Please rename method, it does not check if it is linked, implementation is good as it is but please bring the name in line with the actual implementation